source/mir/stat/distribution/students_t.d

```diff
@@ -0,0 +1,586 @@
+{
+    return mean + stdDev * studentsTInvCDF(p, nu);
+}
+
+///
+version(mir_stat_test)
+@safe pure nothrow @nogc
+unittest {
+    import mir.test: shouldApprox;
+
+    studentsTInvCDF(0.0, 5).shouldApprox == -double.infinity;
+    studentsTInvCDF(0.1, 5).shouldApprox == -1.475884;
+    studentsTInvCDF(0.2, 5).shouldApprox == -0.9195438;
+    studentsTInvCDF(0.3, 5).shouldApprox == -0.5594296;
+    studentsTInvCDF(0.4, 5).shouldApprox == -0.2671809;
+    studentsTInvCDF(0.5, 5).shouldApprox == 0.0;
+    studentsTInvCDF(0.6, 5).shouldApprox == 0.2671809;
+    studentsTInvCDF(0.7, 5).shouldApprox == 0.5594296;
+    studentsTInvCDF(0.8, 5).shouldApprox == 0.9195438;
+    studentsTInvCDF(0.9, 5).shouldApprox == 1.475884;
+    studentsTInvCDF(1.0, 5).shouldApprox == double.infinity;
+
+    // Can include location/scale
+    studentsTInvCDF(0.2, 5, 1, 2).shouldApprox == -0.8390876;
+    studentsTInvCDF(0.4, 5, 1, 2).shouldApprox == 0.4656382;
+    studentsTInvCDF(0.6, 5, 1, 2).shouldApprox == 1.534362;
+    studentsTInvCDF(0.8, 5, 1, 2).shouldApprox == 2.839088;
+}
+
+// Checking other DoF
+version(mir_stat_test)
+@safe pure nothrow @nogc
+unittest {
+    import mir.test: shouldApprox;
+
+    studentsTInvCDF(0.1, 25).shouldApprox == -1.316345;
+    studentsTInvCDF(0.2, 25).shouldApprox == -0.8562362;
+    studentsTInvCDF(0.3, 25).shouldApprox == -0.5311538;
+    studentsTInvCDF(0.7, 25).shouldApprox == 0.5311538;
+    studentsTInvCDF(0.8, 25).shouldApprox == 0.8562362;
+    studentsTInvCDF(0.9, 25).shouldApprox == 1.316345;
+
+    studentsTInvCDF(0.2, double.infinity).shouldApprox == -0.8416212;
+    studentsTInvCDF(0.4, double.infinity).shouldApprox == -0.2533471;
+    studentsTInvCDF(0.6, double.infinity).shouldApprox == 0.2533471;
+    studentsTInvCDF(0.8, double.infinity).shouldApprox == 0.8416212;
+}
+
+// Checking negative location parameter
+version(mir_stat_test)
+@safe pure nothrow @nogc
+unittest {
+    import mir.test: shouldApprox;
+
+    studentsTInvCDF(0.2, 5, -1, 2).shouldApprox == -2.839088;
+    studentsTInvCDF(0.4, 5, -1, 2).shouldApprox == -1.534362;
+    studentsTInvCDF(0.6, 5, -1, 2).shouldApprox == -0.4656383;
+    studentsTInvCDF(0.8, 5, -1, 2).shouldApprox == 0.8390876;
+}
+
+
+/++
+Computes the Student's t log probability distribution function (LPDF).
+
+Params:
+    x = value to evaluate LPDF
+    nu = degrees of freedom
+
+See_also:
+    $(LINK2 https://en.wikipedia.org/wiki/Student%27s_t-distribution, Student's t-distribution)
++/
+@safe pure nothrow @nogc
+T studentsTLPDF(T)(const T x, const T nu)
+    if (isFloatingPoint!T)
+    in (nu > 0, "nu must be greater than zero")
+{
+    import mir.math.common: log, pow, sqrt;
+    import mir.math.internal.xlogy: xlogy;
+    import mir.stat.distribution.normal: normalLPDF;
+    import std.mathspecial: logGamma;
+
+    if (nu != T.infinity) {
+        return logGamma((nu + 1) * 0.5) - logGamma(nu * 0.5) - 0.5 * log(nu) - log(T(SQRTPI)) + xlogy(-(nu + 1) * 0.5, 1 + (x * x) / nu);
```

Comment:
`0.5 * log(nu) - log(T(SQRTPI))` -> `0.5 * log(nu * T(PI))`?

Reply:
Thanks. Good catch. Fixed in https://github.com/libmir/mir-stat/pull/62/commits/75d5fc20fb064d4ddb119e839464f3c4f1784394. Also, `xlogy` isn't needed since `x` can't be zero